--non-interactive flag runtime error

Howdy! 

When you run ast-metrics from the command line and specify the `--non-interactive` or `-i` flags for running static analysis, a runtime error is returned. 

Running: `ast-metrics a -i --report-markdown=report.md ../ConfirmEdit`

One gets:
```
AST Metrics is a language-agnostic static code analyzer.

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x1043b1290]

goroutine 12 [running]:
github.com/pterm/pterm.(*SpinnerPrinter).UpdateText(0x104ca79d0?, {0x140008d7890?, 0x140001d1f18?})
        /home/runner/go/pkg/mod/github.com/pterm/pterm@v0.12.79/spinner_printer.go:121 +0x20
github.com/halleck45/ast-metrics/src/Engine/Python.PythonRunner.DumpAST({0x0, 0x1400019a380, {{0x14000523340, 0x2, 0x2}, 0x14000a30780}})
        /home/runner/work/ast-metrics/ast-metrics/src/Engine/Python/PythonRunner.go:64 +0x158
github.com/halleck45/ast-metrics/src/Command.(*AnalyzeCommand).ExecuteRunnerAnalysis.func1()
        /home/runner/work/ast-metrics/ast-metrics/src/Command/AnalyzeCommand.go:290 +0x34
created by github.com/halleck45/ast-metrics/src/Command.(*AnalyzeCommand).ExecuteRunnerAnalysis in goroutine 1
        /home/runner/work/ast-metrics/ast-metrics/src/Command/AnalyzeCommand.go:289 +0x35c
```

--- a/src/Engine/Python/PythonRunner.go
+++ b/src/Engine/Python/PythonRunner.go
@@ -50,7 +50,9 @@ func (r *PythonRunner) Ensure() error {
 
 // Finish cleans up the workspace
 func (r PythonRunner) Finish() error {
-	r.progressbar.Stop()
+	if r.progressbar != nil {
+		r.progressbar.Stop()
+	}
 	return nil
 }
 
@@ -61,7 +63,9 @@ func (r PythonRunner) DumpAST() {
 	for _, filePath := range r.getFileList().Files {
 
 		cnt++
-		r.progressbar.UpdateText("🐍 Dumping AST of Python files (" + fmt.Sprintf("%d", cnt) + "/" + fmt.Sprintf("%d", len(r.getFileList().Files)) + ")")
+		if r.progressbar != nil {
+			r.progressbar.UpdateText("🐍 Dumping AST of Python files (" + fmt.Sprintf("%d", cnt) + "/" + fmt.Sprintf("%d", len(r.getFileList().Files)) + ")")
+		}
 
 		hash, err := Storage.GetFileHash(filePath)
 		if err != nil {
@@ -80,7 +84,9 @@ func (r PythonRunner) DumpAST() {
 		Engine.DumpProtobuf(protoFile, binPath)
 	}
 
-	r.progressbar.Info("🐍 Python code dumped")
+	if r.progressbar != nil {
+		r.progressbar.Info("🐍 Python code dumped")
+	}
 
 }
 

--- a/src/Engine/Python/PythonRunner_test.go
+++ b/src/Engine/Python/PythonRunner_test.go
@@ -5,6 +5,10 @@ import (
 	"os"
 	"testing"
 
+	"github.com/golang/protobuf/proto"
+	"github.com/halleck45/ast-metrics/src/Configuration"
+	pb "github.com/halleck45/ast-metrics/src/NodeType"
+	"github.com/halleck45/ast-metrics/src/Storage"
 	"github.com/stretchr/testify/assert"
 )
 
@@ -98,4 +102,70 @@ class calculatrice:
 	assert.Equal(t, int32(5), func2.LinesOfCode.LogicalLinesOfCode, "Expected LLOC")
 	assert.Equal(t, int32(0), func2.LinesOfCode.CommentLinesOfCode, "Expected LLOC")
 
+}
+
+func TestParsingPythonFiles(t *testing.T) {
+	pyFile := `
+import functools
+
+class calculatrice:
+	"""
+	A multiline line comment is here
+	A multiline line comment is here
+	"""
+
+	def add(self, a, b):
+		"""
+		A multiline line comment is here
+		A multiline line comment is here
+		"""
+		return a + b`
+
+	// Create a temporary file
+	sourceDirectory := t.TempDir()
+	tmpFile := sourceDirectory + string(os.PathSeparator) + "test.py"
+	defer os.Remove(tmpFile)
+	if _, err := os.Create(tmpFile); err != nil {
+		t.Error(err)
+	}
+	if err := os.WriteFile(tmpFile, []byte(pyFile), 0644); err != nil {
+		t.Error(err)
+	}
+
+	// Configure destination
+	storage := Storage.Default()
+	storage.Purge()
+	storage.Ensure()
+	workdir := storage.AstDirectory()
+
+	// Configure the runner
+	configuration := Configuration.NewConfiguration()
+	configuration.Storage = storage
+	configuration.SetSourcesToAnalyzePath([]string{sourceDirectory})
+	runner := PythonRunner{}
+	runner.SetConfiguration(configuration)
+	runner.DumpAST()
+
+	// list files
+	files, err := os.ReadDir(workdir)
+	if err != nil {
+		t.Error(err)
+	}
+	// check if bin file exists
+	assert.Equal(t, 1, len(files), "Expected 1 file in %s, got %d", workdir, len(files))
+
+	// read the file, and deserialize it to check if it's a protobuf
+	file := files[0]
+	binPath := workdir + string(os.PathSeparator) + file.Name()
+	in, err := os.ReadFile(binPath)
+	if err != nil {
+		t.Error(err)
+	}
+	pbFile := &pb.File{}
+	if err := proto.Unmarshal(in, pbFile); err != nil {
+		t.Error(err)
+	}
+
+	// Ensure path
+	assert.Contains(t, pbFile.Path, "test.py", "Expected path to contain 'test.py', got %s", pbFile.Path)
 }

